Essentials/src/com/earth2me/essentials/utils/LocationUtil.java

```diff
@@ -49,46 +49,39 @@ public static ItemStack convertBlockToItem(final Block block) {
         public int y;
         public int z;
 
-        public Vector3D(int x, int y, int z) {
+        Vector3D(int x, int y, int z) {
             this.x = x;
             this.y = y;
             this.z = z;
         }
     }
 
     static {
-        List<Vector3D> pos = new ArrayList<Vector3D>();
+        List<Vector3D> pos = new ArrayList<>();
         for (int x = -RADIUS; x <= RADIUS; x++) {
             for (int y = -RADIUS; y <= RADIUS; y++) {
                 for (int z = -RADIUS; z <= RADIUS; z++) {
                     pos.add(new Vector3D(x, y, z));
                 }
             }
         }
-        Collections.sort(pos, new Comparator<Vector3D>() {
-            @Override
-            public int compare(Vector3D a, Vector3D b) {
-                return (a.x * a.x + a.y * a.y + a.z * a.z) - (b.x * b.x + b.y * b.y + b.z * b.z);
-            }
-        });
+        pos.sort(Comparator.comparingInt(a -> (a.x * a.x + a.y * a.y + a.z * a.z)));
         VOLUME = pos.toArray(new Vector3D[0]);
     }
 
-    @SuppressWarnings("deprecation")
     public static Location getTarget(final LivingEntity entity) throws Exception {
         Block block = null;
         try {
             block = entity.getTargetBlock(TRANSPARENT_MATERIALS, 300);
-        } catch (NoSuchMethodError e) {
-            // failing now :(
-        }
+        } catch (NoSuchMethodError ignored) {} // failing now :(
         if (block == null) {
             throw new Exception("Not targeting a block");
         }
         return block.getLocation();
     }
 
-    static boolean isBlockAboveAir(final World world, final int x, final int y, final int z) {
+    public static boolean isBlockAboveAir(final World world, final int x, final int y, final int z) {
+        PaperLib.getChunkAtAsync(world, x, z); // Get the chunk first so it can be loaded async
```

Comment:
This call is useless when you simply do getBlockAt in the next line. This function would need to return CompletableFuture<Boolean>.

Reply:
If the chunk isn't loaded, getChunkAtAsync should queue up the chunk load, however the next line then immediately _has_ to load the chunk in sync.